src/fabric/src/fabric2_server.erl

```diff
@@ -137,8 +139,62 @@ init(_) ->
         end,
     application:set_env(fabric, ?FDB_DIRECTORY, Dir),
     config:subscribe_for_changes([?TX_OPTIONS_SECTION]),
+    check_config_limits(),
     {ok, nil}.
 
+check_config_limits() ->
```

Comment:
Looks good, but I think we can simplify it a bit to be less repetitive. Something like this perhaps?

```erlang
-define(CONFIG_LIMITS, [
    {"couchdb", "max_document_size", ?DOCUMENT_SIZE_LIMIT_BYTES},
    {"fabric", "binary_chunk_size", ?DEFAULT_BINARY_CHUNK_SIZE_BYTES},
    ...
]).


check_config_limits() ->
    lists:foreach(fun({Sect, Key, Limit}) ->
        case config:get_integer(Sect, Key, Limit) > Limit of
            true ->
                LogMsg = "Config value of ~p for [~s] ~s is greater than the limit: ~p",
                couch_log:warning(LogMsg, [Val, Sect, Key, Limit]);
            false ->
                ok
        end
    end, ?CONFIG_LIMITS).

```

Reply:
Great simplification! Changed in https://github.com/apache/couchdb/pull/3632/commits/36eb7f16dd8935cd8792a7297aadebe535f109d1.